### ISSUE TITLE
setup.py imports numpy, but numpy is an install time dependency

### ISSUE THREAD

**Issue by mrshannon** [SciTools/cf-units#144]
The `setup.py` file imports `numpy` which is an installation dependency.  Because installation order cannot be specified (in order to ensure `numpy` is installed first), any package that depends on `cf-units` may also fail to install because `numpy` may not exist.

To reproduce the problem create a fresh virtualenv and run `pip install cf-units` which will result in:
```
pip install cf-units
Collecting cf-units
  Using cached https://files.pythonhosted.org/packages/ae/31/d803fd21a960db0cd9d03fd134e3551dfc96a8536956d8309caa4f44db90/cf-units-2.0.2.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-1f2to47t/cf-units/setup.py", line 7, in <module>
        import numpy as np
    ModuleNotFoundError: No module named 'numpy'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-1f2to47t/cf-units/
```

Therefore, `cf-units` is effectively broken for the standard python tool chain.

**Comment by pelson** [comment]
Thanks for the report @mrshannon. Truth be told, the *standard python tool chain* is broken when it comes to depending upon numpy - there is no universal way to define ``setup.py`` time dependencies that works well today (e.g. https://stackoverflow.com/questions/54117786/add-numpy-get-include-argument-to-setuptools-without-preinstalled-numpy). There is promise that this will change with the recent introduction of ``pyproject.toml`` in [PEP518](https://www.python.org/dev/peps/pep-0518/), but even this appears to have some [pretty terrible warts](https://github.com/SciTools/cartopy/pull/1264#issuecomment-454366232) that prohibit uptake.

Given cf-units' dependence on a non-python binary (UDUNITS-2) it seems to me that you are already outside the realms of "standard python tool chain" (you are never going to be able to install cf-units from pypi unless we statically ship UDUNITS-2 with it, which I personally think would be a huge maintenance burden - and the wrong solution).

I know it isn't to everybody's tase, but we have worked hard to make cf-units installable through conda ``conda install -c conda-forge cf-units`` as well as making it *relatively* easy to ``pip install`` once you have ``UDUNITS-2`` and ``numpy`` installed (I recently removed the install-time dependence on Cython for source distributions, but not when cloning from github).

In short, I don't think there is much to be done here. Do you have an alternative suggestion @mrshannon?

**Comment by mrshannon** [comment]
The method in the SO answer that @pelson mentioned does work for `cf-units` as shown in #145.

As for needing a non-python binary, that is not unusual.  For instance [lxml](https://github.com/lxml/lxml) (a popular XML library) requires  `libxml2` to be installed on the system and installs easily from PyPI.

Needing to preinstall `numpy` would not be an issue if I where the final user, but the issue makes building packages that depend on `cf-units` hard to justify.

**Comment by pelson** [comment]
Closed by #145. Thanks for putting the effort in to see this resolved @mrshannon. 👍
